sdk/core/core-client-rest/src/sendRequest.ts

```diff
@@ -37,11 +37,42 @@ export async function sendRequest(
 ): Promise<HttpResponse> {
   const httpClient = customHttpClient ?? getCachedDefaultHttpsClient();
   const request = buildPipelineRequest(method, url, options);
+  if (options) {
+    if (options.timeout) {
+      request.timeout = options.timeout;
+    }
+
+    if (options.onUploadProgress) {
+      request.onUploadProgress = options.onUploadProgress;
+    }
+
+    if (options.onDownloadProgress) {
+      request.onDownloadProgress = options.onDownloadProgress;
+    }
+
+    if (options.allowInsecureConnection) {
+      request.allowInsecureConnection = true;
+    }
+
+    if (options.abortSignal) {
+      request.abortSignal = options.abortSignal;
+    }
+
+    if (options.tracingOptions) {
+      request.tracingOptions = options.tracingOptions;
+    }
+  }
+  
   const response = await pipeline.sendRequest(httpClient, request);
+
   const rawHeaders: RawHttpHeaders = response.headers.toJSON();
 
   const parsedBody: RequestBodyType | undefined = getResponseBody(response);
 
+  if (options?.onResponse) {
+    options.onResponse({...response, request, rawHeaders, parsedBody});
+  }
```

Comment:
In the original core-client onResponse calling, it has the logic of handling the errors, I am not sure if we should add the try catch here, as I think some of the errors are probably throw by deserialization ? not quite sure.

Reply:
Yeah, the deserializationPolicy can throw for core-client, but I don't believe we have any policies that throw in RLC when handling the response.